### PR TITLE
Add delay option to input methods

### LINKED ISSUE #156: Add delay option to input methods
For `mouse.click`, delay should specify the time between `mouse.down` and `mouse.up`.

For `keyboard.press` delay should specify the time between `keyboard.down` and `keyboard.up`.

For `keyboard.type` delay should specify both the time between each key is pressed, and the time between the `down` and `up`.

### PR DESCRIPTION
**Pull request by JoelEinbinder**
Closes #156 

This lets you easily simulate a long press of a key or mouse press, or a user typing at a normal speed.

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -826,6 +830,7 @@ Dispatches a `keyup` event.
 - `options` <[Object]>
   - `button` <[string]> `left`, `right`, or `middle`, defaults to `left`.
   - `clickCount` <[number]> defaults to 1
+  - `delay` <[number]> Time to wait between `mousedown` and `mouseup` in milliseconds. Defaults to 0.
```
Comment (aslushnikov) [inline docs/api.md:838]: Let's add an example which simulates a user typing in one field, and fills instantly another

File: test/test.js [hunk 2]
```diff
@@ -1098,6 +1098,17 @@ describe('Puppeteer', function() {
           fail(modifiers[modifier] + ' should be false');
       }
     }));
+    it('should type with a delay', SX(async function(){
+      await page.navigate(PREFIX + '/input/textarea.html');
+      await page.focus('textarea');
+      await page.type('Hello World!', {delay: 50});
```
Comment (aslushnikov) [inline test/test.js]: this test doesn't test typing with a delay, doesn't it? (it would pass without the changes in `lib/`)

File: test/test.js [hunk 3]
```diff
@@ -1098,6 +1098,17 @@ describe('Puppeteer', function() {
...
+      expect(await page.evaluate(() => document.querySelector('textarea').value)).toBe('Hello World!');
+    }));
+    it('should click with a delay', SX(async function(){
+      await page.navigate(PREFIX + '/input/button.html');
```
Comment (aslushnikov) [inline test/test.js]: looks like this would pass as well

Comment (JoelEinbinder) [reply test/test.js]: It would, it is just a sanity check. Added a comment and made the delay 1.

File: docs/api.md [hunk 4]
```diff
@@ -358,6 +358,7 @@ Shortcut for [page.mainFrame().addScriptTag(url)](#frameaddscripttagurl).
 - `options` <[Object]>
   - `button` <[string]> `left`, `right`, or `middle`, defaults to `left`.
```
Comment (gajus) [inline docs/api.md:359]: What does left, right, middle refer to?

Comment (Urigo) [reply docs/api.md:359]: @gajus if I understand correctly, it's the mouse buttons - `left button click`, `right button click` and `middle button click`

Comment (gajus) [reply docs/api.md:359]: I see. I guess it was not obvious since I haven't used a mouse in the last +5 years. :-)

**Pull request merged**